test/components/fetchers/test_link_content_fetcher.py

```diff
@@ -356,3 +356,50 @@ async def test_run_async_with_client_kwargs(self):
         streams = (await fetcher.run_async([HTML_URL]))["streams"]
         assert len(streams) == 1
         assert "Haystack" in streams[0].data.decode("utf-8")
+
+
+class TestLinkContentFetcher_CustomHeaders:
```

Comment:
No need to wrap this one function in it's own class. Let's just add it to the already made one.